.gitlab/lassen-build-and-test-extra.yml

```diff
@@ -91,45 +91,43 @@ xl_16_1_1_12_gcc_8_3_1_cuda_11_0_2:
 # CPU ONLY
 ##########
 
-clang_11_0_0:
+clang_14_0_5:
   variables:
-    SPEC: " +openmp %clang@11.0.0"
+    SPEC: " +openmp %clang@14.0.5"
   extends: .build_and_test_on_lassen
 
 ##########
 # CUDA
 ##########
 
-clang_11_0_0_gcc_8_3_1_cuda_11_5_0:
+clang_12_0_1_gcc_8_3_1_cuda_11_5_0:
   variables:
-    SPEC: " +openmp +cuda cuda_arch=70 %clang@11.0.0 cxxflags=\"--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1\" cflags=\"--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1\" ^cuda@11.5.0"
+    SPEC: " +openmp +cuda cuda_arch=70 %clang@12.0.1 cxxflags=\"--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1\" cflags=\"--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1\" ^cuda@11.5.0"
```

Comment:
@adrienbernede I'm almost there with this PR. However, the gcc toolchain doesn't look like it's getting passed to CMake. This clang12 build fails because it is using gcc 4.9.3 headers.

Reply:
Nevermind. I restarted it and it passed. Must have been a caching issue or something like that.